File: 0509-fibonacci-number.py
"""
509. Fibonacci Number

Submitted: October 20, 2025

Runtime: 28 ms (beats 96.94%)
Memory: 17.92 MB (beats 22.47%)
"""

class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0
        a = 0
        b = 1
        for _ in range(n-1):
            b += a
            a = b - a
        return b

File: 0509-fibonacci-number.rs
/*
509. Fibonacci Number

Submitted: October 20, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 2.18 MB (beats 41.18%)
*/

impl Solution {
    pub fn fib(n: i32) -> i32 {
        if n == 0 {
            return 0;
        }
        let mut a = 0;
        let mut b = 1;
        for _ in (0..n-1) {
            b += a;
            a = b - a;
        }
        b
    }
}